### DISCUSSION TITLE
Would there be interest in building a standalone sensor board?

### DISCUSSION THREAD

**Discussion by kubark42** [iltis42/XCVario#239] [Ideas]
It seems that the pitot-static part of this system is timeless, whereas the front end will forever be undergoing hardware improvements (larger screen, brighter screen, more buttons, etc...). Would it be reasonable to create a sensor module which grabs the sensor data, does whatever rudimentary pre-processing is necessary, and then forwards this data on to the front-end? Publishing via UART and CAN would certainly work, ESP-NOW and ESP-WIFI-MESH would be very promising, and publishing via Bluetooth and BLE are probably reasonable approaches as well. (Certainly the variometer data rate doesn't need to exceed 10Hz and 50ms of jitter is likely more than acceptable.)

The advantage is that you decouple the hard work of sensing things from the equally hard work of displaying things. But once those two are decoupled, it slashes prices for future front-end development, and it lets sensor-side development proceed at its own rate. Novel interfaces based on systems such as the https://www.lilygo.cc/products/t-rgb or the https://github.com/scottbez1/smartknob become easily feasible. Or even a pure Android-based interface, if someone were so inclined. And if new sensors hit the market which were more accurate or more cost effective, these would not imply redoing the instrument panel.

Of course, this would be possible right now by simply taking an existing module and shutting it up somewhere in the bowls of the plane. This results in a the screen which is no longer visible, but that's not an enormous hit. Surely the display can be depowered so it's no longer draining any of the glider's limited energy budget. The upshot is the firmware feature set could be fleshed out without building new hardware, and only once it's ready would development focus shift to hardware updates.

**Comment by iltis42** [comment]
Its a good idea, until now XCVario was dedicated as a standalone instrument so far. The display board in this design is already separate, so if there is demand for a standalone board, you may just use that in a standard ABS case.

**Comment by hjr** [comment]
Yes, think so too. The XCV board is quiet headless. A front cover w/o display would yield in a sensor only version of XCV.
